cdap-api-common/src/main/java/io/cdap/cdap/internal/io/SchemaTypeAdapter.java

```diff
@@ -86,6 +86,8 @@ private Schema read(JsonReader reader, Map<String, Schema> knownRecords) throws
     JsonToken token = reader.peek();
     switch (token) {
       case NULL:
+        // advance the value otherwise gson won't be able to get over the null value
+        reader.nextNull();
```

Comment:
was this case never being hit before? Why is it being hit now?

Reply:
This is hit by https://github.com/cdapio/cdap/blob/develop/cdap-app-templates/cdap-etl/cdap-etl-proto/src/main/java/io/cdap/cdap/etl/proto/v2/spec/StageSpec.java#L46-L47 when I add the unit test for validation handler, basically if these two fields are null, I am able to do` GSON.toJson(stagespec)` but not able to transfer it back `GSON.fromJson(stagespecjson, StageSpec.class)` and gson complains it encountered a null while expecting a variable name.